src/Dialogs/Frequency/dlgFrequencyEdit.cpp

```diff
@@ -0,0 +1,41 @@
+#include "dlgFrequencyEdit.hpp"
+
+FrequencyEditResult
+dlgFrequencyEditShowModal(FrequencyListItem &freq)
+{
+    const DialogLook &look = UIGlobals::GetDialogLook();
+    TWidgetDialog<FrequencyEditWidget>
+        dialog(WidgetDialog::Auto{}, UIGlobals::GetMainWindow(),
+               look, _("Frequency Editor"));
+    dialog.AddButton(_("OK"), mrOK);
+    dialog.AddButton(_("Cancel"), mrCancel);
+    dialog.SetWidget(look, freq);
+    const int result = dialog.ShowModal();
+
+    if (result != mrOK)
+        return FrequencyEditResult::CANCEL;
+
+    if (!dialog.GetChanged())
+        return FrequencyEditResult::UNMODIFIED;
+
+    freq = dialog.GetWidget().GetValue();
+    return FrequencyEditResult::MODIFIED;
+}
+
+void FrequencyEditWidget::Prepare(ContainerWindow &, const PixelRect &) noexcept
+{
+    AddText(_("Name"), nullptr, value.name.c_str(), this);
+    AddText(_("Short Name"), nullptr, value.short_name.c_str(), this);
+    AddFloat(_("Frequency"), nullptr, _("%.3f"), _("%.3f"), 118.000, 137.000, 0.005, false, value.freq.GetKiloHertz() / 1000.0, this);
```

Comment:
To make it fully clean we should add an AddFrequency() in EditRowFormWidget  that deals with frequencies.
Then add a check in RadioFrequency to parse the vaildify of the Frequency.